lightapi/core.py

```
@@ -336,3 +467,134 @@ def process(self, request, response):
+
+class CORSMiddleware(Middleware):
+    """
+    CORS (Cross-Origin Resource Sharing) middleware.
+    
+    Handles CORS preflight requests and adds appropriate headers to responses.
+    This provides a more flexible alternative to Starlette's built-in CORS middleware.
+    """
+    
+    def __init__(
+        self, 
+        allow_origins=['*'], 
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)